apps/hash-frontend/src/pages/new/types/entity-type.page.tsx

```diff
@@ -96,14 +120,50 @@ const Page: NextPageWithLayout = () => {
       kind: "entity-type",
       version: 1,
     });
+
     const entityType: EntityType = {
       $schema: ENTITY_TYPE_META_SCHEMA,
       kind: "entityType",
       $id: versionedUrl,
       title: name,
       description,
       type: "object",
-      properties: {},
+      properties: await Promise.all(
+        initialPropertyTypes.map(async (propertyTypeDefinition) => {
+          if ("$id" in propertyTypeDefinition) {
+            return propertyTypeDefinition;
+          }
+
+          const res = await createPropertyType({
+            data: {
+              propertyType: getPropertyTypeSchema({
+                name: propertyTypeDefinition.title,
+                description: propertyTypeDefinition.description,
+                expectedValues: [
+                  types.dataType[propertyTypeDefinition.dataType].dataTypeId,
+                ],
+                flattenedCustomExpectedValueList: {},
+              }),
+            },
+          });
+
+          if (res.errors?.length || !res.data) {
+            /** @todo: handle this */
+
+            return [];
+          }
+
+          return res.data.schema;
+        }),
+      ).then((createdPropertyTypes) =>
+        createdPropertyTypes.flat().reduce(
+          (prev, { $id }) => ({
+            ...prev,
+            [extractBaseUrl($id)]: { $ref: $id },
+          }),
+          {},
+        ),
```

Comment:
Object.fromEntries would be more readable than a reduce

```suggestion
        Object.fromEntries(
          createdPropertyTypes.flat().map(({ $id }) => [
            extractBaseUrl($id),
            { $ref: $id },
          ])
        ),
```